arrows/klv/klv_parse.cxx

```diff
@@ -2,433 +2,479 @@
 // OSI-approved BSD 3-Clause License. See top-level LICENSE file or
 // https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
 
-/**
- * \file
- * \brief This file contains the implementation for the klv parser.
- */
+/// \file
+/// \brief This file contains the implementation for the klv parser.
 
-#include "klv_parse.h"
+#include "klv_0104.h"
+#include "klv_0601.h"
 #include "klv_data.h"
 #include "klv_key.h"
-#include "klv_0601.h"
-#include "klv_0104.h"
+#include "klv_parse.h"
 
 #include <vital/exceptions/metadata.h>
 
 #include <vital/logger/logger.h>
 
+#include <iomanip>
+#include <numeric>
+#include <sstream>
+
 #include <cctype>
 
+namespace kv = kwiver::vital;
+
 namespace kwiver {
+
 namespace arrows {
+
 namespace klv {
 
 namespace {
 
+// ---------------------------------------------------------------------------
```

Comment:
Is this new code?

Reply:
It's lines 113-118 made into a more generalized template function which can work with any iterator or output type. Reading a serialized unisgned integer from a buffer of bytes is a common task when reading KLV ( e.g. code is duplicated almost verbatim in `arrows/klv/klv_0601.cxx:43:48`. I imagine this would get pulled out into some sort of utility file eventually.